fix(playground): pass transform target to minifier

camc314 opened the pull request:
The playground minifier was not respecting the transform target option, causing ES features like optional chaining to be used even when targeting older browsers that do not support them.

Closes #16889

sapphi-red reviewed (commented):
It feels a bit off to respect the options in `transform` section, but I guess that's fine for now.

camc314 commented on napi/playground/src/lib.rs:
```diff
@@ -318,7 +318,16 @@ impl Oxc {
+                let mut compress_options = CompressOptions::smallest();
+                if let Some(transform_options) = &options.transformer
+                    && let Some(target) = &transform_options.target
+                    && let Ok(targets) = oxc_compat::EngineTargets::from_target(target)
```
good call

camc314 committed:
fix(playground): pass transform target to minifier (#16910)

The playground minifier was not respecting the transform target option, causing ES features like optional chaining to be used even when targeting older browsers that do not support them.

Closes #16889
